sqldelight-gradle-plugin/src/test/fixtures/nullability/expected/com/sample/Test2Model.java

```diff
@@ -0,0 +1,334 @@
+    @NonNull
+    T1 test2();
+
+    @Nullable
+    V2 view1();
+  }
+
+  interface Left_join_viewCreator<T1 extends Test2Model, V2 extends View1Model, T extends Left_join_viewModel<T1, V2>> {
+    T create(@NonNull T1 test2, @Nullable V2 view1);
+  }
+
+  final class Left_join_viewMapper<T1 extends Test2Model, V2 extends View1Model, T extends Left_join_viewModel<T1, V2>> implements RowMapper<T> {
+    private final Left_join_viewCreator<T1, V2, T> creator;
+
+    private final Factory<T1> test2ModelFactory;
+
+    private final View1Creator<V2> view1Creator;
+
+    Left_join_viewMapper(Left_join_viewCreator<T1, V2, T> creator, Factory<T1> test2ModelFactory, View1Creator<V2> view1Creator) {
+      this.creator = creator;
+      this.test2ModelFactory = test2ModelFactory;
+      this.view1Creator = view1Creator;
+    }
+
+    @Override
+    @NonNull
+    public T map(@NonNull Cursor cursor) {
+      return creator.create(
+          test2ModelFactory.creator.create(
+              cursor.getLong(0),
+              cursor.isNull(1) ? null : cursor.getLong(1),
+              cursor.getLong(2),
+              cursor.getLong(3)
+          ),
+          cursor.isNull(4) && cursor.isNull(5)
+          ? null
+          : view1Creator.create(
+              cursor.isNull(4) ? null : cursor.getLong(4),
+              cursor.getLong(5)
+          )
+      );
+    }
+  }
+
+  interface Left_join_view_columnsModel<T1 extends Test2Model> {
+    @NonNull
+    T1 test2();
+
+    @Nullable
+    Long nullable_int();
+
+    long nonnull_int();
+  }
+
+  interface Left_join_view_columnsCreator<T1 extends Test2Model, T extends Left_join_view_columnsModel<T1>> {
+    T create(@NonNull T1 test2, @Nullable Long nullable_int, long nonnull_int);
+  }
+
+  final class Left_join_view_columnsMapper<T1 extends Test2Model, T extends Left_join_view_columnsModel<T1>> implements RowMapper<T> {
+    private final Left_join_view_columnsCreator<T1, T> creator;
+
+    private final Factory<T1> test2ModelFactory;
+
+    Left_join_view_columnsMapper(Left_join_view_columnsCreator<T1, T> creator, Factory<T1> test2ModelFactory) {
+      this.creator = creator;
+      this.test2ModelFactory = test2ModelFactory;
+    }
+
+    @Override
+    @NonNull
+    public T map(@NonNull Cursor cursor) {
+      return creator.create(
+          test2ModelFactory.creator.create(
+              cursor.getLong(0),
+              cursor.isNull(1) ? null : cursor.getLong(1),
+              cursor.getLong(2),
+              cursor.getLong(3)
+          ),
+          cursor.isNull(4) ? null : cursor.getLong(4),
+          cursor.getLong(5)
+      );
+    }
+  }
+
+  interface View1Model {
+    @Nullable
+    Long nullable_int();
+
+    long nonnull_int();
+  }
+
+  interface View1Creator<T extends View1Model> {
+    T create(@Nullable Long nullable_int, long nonnull_int);
+  }
+
+  interface Creator<T extends Test2Model> {
+    T create(long _id, Long nullable_int, long nonnull_int, long test2_id);
```

Comment:
`nullable_int` should be annotated with `@Nullable` like other various arguments.

That said, you can omit `@Nullable` on arguments if you want (they're nullable by default). As long as they're on methods with nullable return values that's the most important thing. Since this is generated code I don't think it's a big deal to have them on arguments though.